fault-remediation/pkg/reconciler/reconciler_e2e_test.go

```diff
@@ -916,291 +1020,6 @@ func cleanupNodeAnnotations(ctx context.Context, t *testing.T, nodeName string)
-	defer func() {
-		testClient.CoreV1().Nodes().Delete(ctx, nodeName, metav1.DeleteOptions{})
-	}()
-
-	cleanupNodeAnnotations(ctx, t, nodeName)
-
-	remediationClient := createTestRemediationClient(t, false)
-
-	cfg := ReconcilerConfig{
-		RemediationClient: remediationClient,
-		StateManager:      statemanager.NewStateManager(testClient),
-		UpdateMaxRetries:  3,
-		UpdateRetryDelay:  100 * time.Millisecond,
-	}
-	reconcilerInstance := NewReconciler(cfg, false)
-
-	mockWatcher := storewatcher.NewFakeChangeStreamWatcher()
-	mockColl := &MockCollection{
-		updateOneFn: func(ctx context.Context, filter interface{}, update interface{}, opts ...*options.UpdateOptions) (*mongo.UpdateResult, error) {
-			return &mongo.UpdateResult{ModifiedCount: 1}, nil
-		},
-	}
-
-	reconcilerDone := make(chan struct{})
-	go func() {
-		defer close(reconcilerDone)
-		mockWatcher.Start(ctx)
-		for event := range mockWatcher.Events() {
-			reconcilerInstance.processEvent(ctx, event, mockWatcher, mockColl)
-		}
-	}()
-
-	t.Log("Send Quarantined event to create CR and annotation")
-	eventID1 := primitive.NewObjectID()
-	event1 := createQuarantineEvent(eventID1, nodeName, protos.RecommendedAction_RESTART_BM)
-	mockWatcher.EventsChan <- event1
-
-	var crName string
-	require.Eventually(t, func() bool {
-		state, err := reconcilerInstance.annotationManager.GetRemediationState(ctx, nodeName)
-		if err != nil {
-			return false
-		}
-		if grp, ok := state.EquivalenceGroups["restart"]; ok {
-			crName = grp.MaintenanceCR
-			return crName != ""
-		}
-		return false
-	}, 5*time.Second, 100*time.Millisecond, "CR and annotation should be created")
-
-	t.Log("Send Cancelled event to remove group from annotation")
-	cancelledEvent := bson.M{
-		"operationType": "update",
-		"fullDocument": bson.M{
-			"_id": eventID1,
-			"healtheventstatus": bson.M{
-				"nodequarantined": model.Cancelled,
-			},
-			"healthevent": bson.M{
-				"nodename":          nodeName,
-				"recommendedaction": int32(protos.RecommendedAction_RESTART_BM),
-			},
-		},
-	}
-	mockWatcher.EventsChan <- cancelledEvent
-
-	t.Log("Verify group removed from annotation")
-	require.Eventually(t, func() bool {
-		state, err := reconcilerInstance.annotationManager.GetRemediationState(ctx, nodeName)
-		if err != nil {
-			return false
-		}
-		_, hasRestart := state.EquivalenceGroups["restart"]
-		return !hasRestart
-	}, 5*time.Second, 100*time.Millisecond, "Restart group should be removed from annotation after cancellation")
-
-	close(mockWatcher.EventsChan)
-	<-reconcilerDone
-
-	gvr := schema.GroupVersionResource{
-		Group:    "janitor.dgxc.nvidia.com",
-		Version:  "v1alpha1",
-		Resource: "rebootnodes",
-	}
-	testDynamic.Resource(gvr).Delete(ctx, crName, metav1.DeleteOptions{})
-}
-
-// TestReconciler_CancelledEventClearsAllGroups validates that Cancelled events clear all remediation state
-func TestReconciler_CancelledEventClearsAllGroups(t *testing.T) {
```

Comment:
Is this test not working post refactor? Can we instead add a comment to fix later rather removing?

Reply:
added back